plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/util/HudiAvroSerializer.java

```diff
@@ -207,13 +173,61 @@ else if (javaType == long.class) {
                 else if (type.equals(INTEGER)) {
                     type.writeLong(output, ((Number) value).intValue());
                 }
+                else if (type.equals(SMALLINT)) {
+                    type.writeLong(output, ((Number) value).shortValue());
+                }
+                else if (type.equals(TINYINT)) {
+                    type.writeLong(output, ((Number) value).byteValue());
+                }
+                else if (type.equals(REAL)) {
```

Comment:
Could we add unit tests of `appendTo` on all different branches here?

Reply:
Tried spending an hour on this, requires quite abit of work. Let me add this as an action item to refactor so that it's more testable.

Follow up jira task:
https://issues.apache.org/jira/browse/HUDI-9434

Reply:
Sg